### PR TITLE
Update Readme after first pass of setup

### PR DESCRIPTION
**Pull request by TessFerrandez**
- Clean up typos
- General rewording of steps - clarification etc. to make it a bit easier to go through
- Added todos comments in the draft - these will be removed before merging the PR.

### CODE REVIEW

**Comment by sachinkundu** [comment]
Looks good! Let's merge when done and go from there. Let's ask someone else to run through it and revaluate after that.

File: setup.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+> **IMPORTANT:** There is a cost associated with provisioning resources on Azure. Specifically the Azure Firewall costs by the hour. If you are setting this up for testing purposes, you may want to tear down all resources after testing to avoid high bills for resources you don't use.
+
+## Software requirements
+
+We have tested this repo with OSX, Linux and Windows (on [WSL](https://docs.microsoft.com/en-us/windows/wsl/)).
+
+You need to install:
+
+* [curl](https://curl.se/)
+* [kubectl min version 1.18.1](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
```
Comment (Lybecker) [inline setup.md:12]: This step is not needed anymore, as it is installed via AZ cli

File: setup.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+You can also use this [Docker dev container](./utils/docker-dev-env/README.md) if you don't want to install the tools on your own machine.
+
+> **NOTE:** The installation assumes that you have an Azure Active Directory, and that you are the owner/admin of this directory or can request to create AAD apps. We will need this for RBAC setup for Kubernetes. For testing purposes you can [create a personal AAD](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-access-create-new-tenant).
+
+## One time setup
+
+Before provisioning the AKS resources we need to prepare some repositories, set up AAD security groups and Service Principals.
+
+[TODO]: # (is this correct?)
```
Comment (Lybecker) [inline setup.md:34]: Yes. We need that stuff. @sachinkundu can you describe what is needed?

File: setup.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+Terraform stores state configuration in Azure Storage.
+
+1. If you don't have a storage account to use for terraform state already configured. Create a resource group and a storage account with a container per the instructions from the Microsoft docs [Tutorial: Store Terraform state in Azure Storage](https://docs.microsoft.com/en-us/azure/developer/terraform/store-state-in-azure-storage)
+
+    ```bash
+    #!/bin/bash
+
+    RESOURCE_GROUP_NAME=tstate
+    STORAGE_ACCOUNT_NAME=tstate$RANDOM
```
Comment (Lybecker) [inline setup.md:108]: Where does the @RANDOM comes from?

File: setup.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+    CONTAINER_NAME=tstate
+
+    # Create resource group
+    az group create --name $RESOURCE_GROUP_NAME --location eastus
```
Comment (Lybecker) [inline setup.md:112]: Hard-coded location

File: setup.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+3. Execute the terraform scripts to provision the resources (from the [1-preprovision](./1-preprovision/) folder)
+
+    ```bash
+    terraform init
+    terraform plan
+    terraform apply
+    ```
+
+> NOTE: It's normal for this to take a long time to provision, especially the Firewall, so relax and grab a coffee.
```
Comment (Lybecker) [inline setup.md:216]: I ❤️ this :)

File: setup.md [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+
+## Provision an AKS cluster
+
+1. Create a custom Grafana image (from the [./utils/grafana/](./utils/grafana/) folder)
+
+    ```bash
+    export registry=registry_name_here
```
Comment (Lybecker) [inline setup.md:223]: Description: registry_name_here var is the name of the Azure Container Registry created during pre-provisioning.

File: setup.md [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+    MSI_RESOURCE_ID=$(az identity show -n magicaksmsi -g $RG_WHERE_CLUSTER_EXISTS -o json | jq -r ".id")
+    az role assignment create --role "Network Contributor" --assignee $MSI_CLIENT_ID -g $RG_WHERE_NETWORK_EXISTS
+    az role assignment create --role "Virtual Machine Contributor" --assignee $MSI_CLIENT_ID -g $RG_WHERE_NETWORK_EXISTS
+    ```
+
+    > **NOTE** MagicAKS is not creating a system assigned managed identity, due to current [limitations](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity#create-an-aks-cluster-with-managed-identities) of self managed VNet and static IP address outside the MC_ resource group.
+
+    We provide ``MSI_RESOURCE_ID`` as a variable to terraform when creating the cluster.
+
+    [TODO]: # (figure out what this is and which variable we are talking about)
```
Comment (Lybecker) [inline setup.md:246]: MSI_RESOURCE_ID is the Managed Identity created above. The AKS cluster will use this identity when accsing other resource like ACR, Key Vault etc.

**Review by Lybecker — APPROVED** [review]

File: setup.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+* [fluxctl](https://docs.fluxcd.io/en/1.18.0/references/fluxctl.html)
+
+  To install fluxctl on WSL run
+
+    ```bash
+    sudo curl -L https://github.com/fluxcd/flux/releases/download/1.21.1/fluxctl_linux_amd64 -o
+    /usr/local/bin/fluxctl
+    ```
+
+* [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) logged in with a user with permissions to provision resources and create applications on the AAD.
```
Comment (tompaana) [inline setup.md:21]:
```suggestion
* [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) [logged in](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli) with a user with permissions to provision resources in Azure and manage [Azure Active Directory (AAD)](https://azure.microsoft.com/en-us/services/active-directory/).
```

File: setup.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+* [Terraform](https://www.terraform.io/downloads.html) Tested with v0.14.2
```
Comment (tompaana) [inline setup.md:22]:
```suggestion
* [Terraform](https://www.terraform.io/downloads.html) (tested with version 0.14.2)
```

File: setup.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+* [jq](https://stedolan.github.io/jq/)
+
+You can also use this [Docker dev container](./utils/docker-dev-env/README.md) if you don't want to install the tools on your own machine.
```
Comment (tompaana) [inline setup.md:25]:
```suggestion
You can also use this [Docker dev container](./utils/docker-dev-env/README.md), if you don't want to install the tools in your native environment, but note that there can be additional considerations using the Docker development environment that are not covered by this guide.
```

File: setup.md [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+
+> **NOTE:** The installation assumes that you have an Azure Active Directory, and that you are the owner/admin of this directory or can request to create AAD apps. We will need this for RBAC setup for Kubernetes. For testing purposes you can [create a personal AAD](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-access-create-new-tenant).
```
Comment (tompaana) [inline setup.md:27]:
```suggestion
> **Note:** The installation assumes that you have an Azure Active Directory (AAD), and that you are the owner/admin of this directory or can request to create AAD apps. We will need this for [role-based access control (RBAC)](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview) setup for Kubernetes. For testing purposes you can [create a personal AAD](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-access-create-new-tenant).
```

File: setup.md [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+
+## One time setup
+
+Before provisioning the AKS resources we need to prepare some repositories, set up AAD security groups and Service Principals.
```
Comment (tompaana) [inline setup.md:31]:
```suggestion
Before provisioning the AKS resources we need to prepare some repositories, set up AAD security groups and service principals.
```

File: setup.md (hunk 2 above)
Comment (TessFerrandez) [reply setup.md:34]: ahh, the "is this correct was for the line below" i.e. "you can do everything in this section once, and reuse the assets when spinning up new AKS clusters"

File: setup.md [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+
+[TODO]: # (is this correct?)
+
+You can do everything in this section once, and reuse the assets when spinning up new AKS clusters.
+
+### 1. Duplicate this repository
+
+[Duplicate](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/duplicating-a-repository) this repository in GitHub and make your repository private. You will need to change some of the terraform scripts etc. throughout this setup.
```
Comment (tompaana) [inline setup.md:39]:
```suggestion
[Duplicate](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/duplicating-a-repository) this repository in GitHub and make your repository private. You will need to change some of the Terraform scripts etc. throughout this setup.
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -1,30 +1,27 @@
...
-#### **Do you intend to add a new feature or change an existing one?**
+## **Do you intend to add a new feature or change an existing one?**
 
-* Open a [new issue](https://github.com/magicaks/magicaks/issues/new) so we can first dicuss the changes there. After the discussion you may send a PR with relevant changes. Do not send a PR without discussion on an issue.
+* Open a [new issue](https://github.com/magicaks/magicaks/issues/new) so we can first discuss the changes there. After the discussion you may send a PR with relevant changes. Do not send a PR without discussion on an issue.
 
-#### **Do you have questions about the source code?**
+## **Do you have questions about the source code?**
 
 * Ask [Sachin Kundu](mailto:skundu@hey.com)
```
[the remark this answers is not in the document]
Comment (TessFerrandez) [reply CONTRIBUTING.md:22]: I will remove this part for now - since we already have a section about opening a new issue

File: setup.md [hunk 15]
```diff
@@ -0,0 +1,304 @@
...
+
+### 2. Create an AKS cluster admins AAD group
+
+1. Log in as AAD admin.
```
Comment (tompaana) [inline setup.md:43]:
```suggestion
1. Log in as AAD admin:
```

File: README.md [hunk 16]
```diff
@@ -1,179 +1,75 @@
...
+Planned components
 
-> **NOTE** Name of the cluster acts as a prefix for many other resources created and hence to fulfill Azure naming requirements it is essential to select a shorter name.
+* [Azure SQL](https://docs.microsoft.com/en-us/azure/azure-sql/azure-sql-iaas-vs-paas-what-is-overview) as database provider.
+* [Azure Application Gateway](https://docs.microsoft.com/en-us/azure/application-gateway/features) as ingress controller.
 
-* There is a folder ``1-preprovision`` which contains terraform scripts to create those resources which need to be created just onetime. For example vnet, subnets azure firewall and its rules and keyvault etc. After you have verified all variable names you can exectute terraform.
+When needed, the solution uses open source tooling like [kured](https://github.com/weaveworks/kured).
```
[the remark this answers is not in the document]
Comment (TessFerrandez) [reply README.md:43]: I will add an issue to look over the tools we use and add them

File: setup.md [hunk 17]
```diff
@@ -0,0 +1,304 @@
...
+
+    ```sh
+    az login
+    ```
+
+    > NOTE: You need to log in on tenant that you want to manage RBAC from - to login in to a specific tenant use `az login --tenant <tenant_id>` and if your AAD tenant does not have a subscription, run `az login --tenant <tenant_id> --allow-no-subscriptions` instead.
```
Comment (tompaana) [inline setup.md:49]:
```suggestion
    > **Note:** You need to log in on tenant that you want to manage RBAC from - to login into a specific tenant use `az login --tenant <tenant_id>` and if your AAD tenant does not have a subscription, run `az login --tenant <tenant_id> --allow-no-subscriptions` instead.
```

File: setup.md [hunk 18]
```diff
@@ -0,0 +1,304 @@
...
+
+2. Create a new Azure AD group for the AKS cluster admins
```
Comment (tompaana) [inline setup.md:51]:
```suggestion
2. Create a new Azure AD group for the AKS cluster admins:
```

File: setup.md (hunk 15 above)
Comment (tompaana) [inline setup.md:43]: Nit: Perhaps just stick to using "1." for all the items in the ordered list (unless something in the Markdown breaks the automatic numbering).

File: setup.md [hunk 19]
```diff
@@ -0,0 +1,304 @@
...
+
+    ```sh
+    az ad group create --display-name magicaksadmins --mail-nickname magicaksadmins
+    ```
+
+3. Write down the **Object ID** of the group as you will need to supply this when creating the AKS cluster.
+
+### 3. Set up the admin repository and configure RBAC
+
+1. Duplicate the [fabricate-defs repository](https://github.com/magicaks/fabrikate-defs). We recommended making this repository private.
```
Comment (tompaana) [inline setup.md:61]:
```suggestion
1. Duplicate the [Fabrikate high-level definitions (HLD) repository](https://github.com/magicaks/fabrikate-defs). We recommended making this repository private. You will use this repository, for example, to generate RBAC and Azure Monitor configuration.
```

File: setup.md [hunk 20]
```diff
@@ -0,0 +1,304 @@
...
+
+    This will be your Fabrikate High Level Definition (HLD) repo.
```
Comment (tompaana) [inline setup.md:63]:
```suggestion
```

File: setup.md [hunk 21]
```diff
@@ -0,0 +1,304 @@
...
+
+    You will use this repository to generate RBAC and Azure Monitor configuration
```
Comment (tompaana) [inline setup.md:65]:
```suggestion
```

File: setup.md [hunk 22]
```diff
@@ -0,0 +1,304 @@
...
+
+2. Follow the steps in the [README](https://github.com/magicaks/fabrikate-defs/blob/master/README.md) to set up RBAC for your cluster.
```
Comment (tompaana) [inline setup.md:67]:
```suggestion
2. Follow the steps in the [Fabrikate definitions repository README](https://github.com/magicaks/fabrikate-defs/blob/master/README.md) to set up RBAC for your cluster.
```

File: setup.md [hunk 23]
```diff
@@ -0,0 +1,304 @@
...
+
+> **NOTE:** Make sure you finish the steps in the README before you continue
```
Comment (tompaana) [inline setup.md:69]:
```suggestion
> **Note:** Make sure you finish the steps in the README before you continue.
```

File: README.md [hunk 24]
```diff
@@ -1,179 +1,75 @@
...
-This automation is divided in 3 stages. We use terraform to bootstrap infrastructure and set up a gitOps connection in the cluster which then picks up kubernetes manifests for the declared state of the cluster.
-
-These manifests are in turn divided into two parts.
-
-1. Manifests which require admin permissions on the cluster to install.
-2. Manifests for workloads which run within a namespace and with limited credentials.
-
-This is done to prevent unauthorized installations in the cluster. Hence, there are two flux pods running, one for each type of manifest. Admin manifests should come from a repository which is controlled by cluster admins and has tighter process control methods so that every input is checked before being applied to the cluster.
-
-Instead of writing k8s manifests directly we use Fabrikate HLD to write the config which is then translated to K8s manifests using a pipeline which in turn runs ``fab generate`` and pushes the config to the k8s manifest repo. However you do not need to use Fabrikate. The automation expects names of kubernetes manifest repo and you can use any manifest generation tool of your choice, for example kustomize.
```
[the remark this answers is not in the document]
Comment (TessFerrandez) [reply README.md:44]: Removed this

File: setup.md [hunk 25]
```diff
@@ -0,0 +1,304 @@
...
+
+### 4. Set up the user workloads manifest repository
+
+1. Duplicate the [k8sworkloads repository](https://github.com/magicaks/k8sworkloads). We recommended making this repository private.
+
+This is user workloads manifest repo where you list non-privileged workloads. Flux gitOps non admin controller will track this repository.
```
Comment (tompaana) [inline setup.md:75]:
```suggestion
This is user workloads manifest repo where you list non-privileged workloads. [Flux (GitOps)](https://fluxcd.io/) non-admin controller will track this repository.
```

File: setup.md [hunk 26]
```diff
@@ -0,0 +1,304 @@
...
+
+### 5. Create Service Principals for provisioning resources
```
Comment (tompaana) [inline setup.md:77]:
```suggestion
### 5. Create service principals for provisioning resources
```

File: setup.md [hunk 27]
```diff
@@ -0,0 +1,304 @@
...
+
+1. Create a service principal (**terraform-magicaks**) that terraform can use for deploying resources. **Write down the id and password for later.**
```
Comment (tompaana) [inline setup.md:79]:
```suggestion
1. Create a service principal (**terraform-magicaks**) that Terraform can use for deploying resources and make sure to **write down the ID and password for later**:
```

File: setup.md [hunk 28]
```diff
@@ -0,0 +1,304 @@
...
+
+    > **NOTE:** Replace SUBSCRIPTION_ID with your own Azure subscription ID
```
Comment (tompaana) [inline setup.md:81]:
```suggestion
    > **Note:** Replace `SUBSCRIPTION_ID` with your own Azure subscription ID.
```

File: setup.md [hunk 29]
```diff
@@ -0,0 +1,304 @@
...
+
+    ```bash
+    az ad sp create-for-rbac --role="Contributor" --name "terraform-magicaks" --scopes="/subscriptions/SUBSCRIPTION_ID"
+    OBJECT_ID=$(az ad sp show --id app_id_from_above --query objectId)
+    az role assignment create --assignee-object-id $OBJECT_ID --role "Resource Policy Contributor" # Needed to assign Azure Policy to cluster.
+    ```
+
+    [TODO]: # (why does this need --scopes and not the next one)
+
+2. Create a service principal (**magicaks-grafana**) that grafana can use for talking to log analytics backend. We restrict this service principal to **Monitoring Reader**.
```
Comment (tompaana) [inline setup.md:91]:
```suggestion
2. Create a service principal (**magicaks-grafana**) that Grafana can use for talking to Log Analytics backend. We restrict this service principal to **Monitoring Reader** role.
```

File: setup.md (hunk 3 above)
Comment (TessFerrandez) [reply setup.md:108]: $RANDOM gives a random number - so the storage is unique

File: setup.md (hunk 4 above)
Comment (TessFerrandez) [reply setup.md:112]: Fixing

File: setup.md (hunk 27 above)
Comment (tompaana) [inline setup.md:79]: Note for later: We should be consistent with naming and change this to "magicaks-terraform".

File: setup.md [hunk 30]
```diff
@@ -0,0 +1,304 @@
...
+
+    ```bash
+    az ad sp create-for-rbac  --role "Monitoring Reader" --name "magicaks-grafana"
+    ```
+
+### 6. Configure Terraform state
+
+Terraform stores state configuration in Azure Storage.
+
+1. If you don't have a storage account to use for terraform state already configured. Create a resource group and a storage account with a container per the instructions from the Microsoft docs [Tutorial: Store Terraform state in Azure Storage](https://docs.microsoft.com/en-us/azure/developer/terraform/store-state-in-azure-storage)
```
Comment (tompaana) [inline setup.md:101]:
```suggestion
1. If you don't have a storage account to use for Terraform state already configured, create a resource group and a storage account with a container per the instructions from the Microsoft docs: [Tutorial: Store Terraform state in Azure Storage](https://docs.microsoft.com/en-us/azure/developer/terraform/store-state-in-azure-storage):
```

File: setup.md (hunk 6 above)
Comment (TessFerrandez) [reply setup.md:213]: adding this

File: setup.md (hunk 7 above)
Comment (TessFerrandez) [reply setup.md:236]: fixing

File: setup.md [hunk 31]
```diff
@@ -0,0 +1,304 @@
...
+    az storage container create --name $CONTAINER_NAME --account-name $STORAGE_ACCOUNT_NAME --account-key $ACCOUNT_KEY
+
+    echo "storage_account_name: $STORAGE_ACCOUNT_NAME"
+    echo "container_name: $CONTAINER_NAME"
+    echo "access_key: $ACCOUNT_KEY"
+    ```
+
+2. Note the storage account name, container name and storage access key. You will need the storage access key in step 7.
+
+3. Update the terraform section of [1-preprovision/main.tf](1-preprovision/main.tf) with your values for resource_group_name, container_name, and storage_account_name. Leave the key as **"magicaks-longlasting"**.
```
Comment (tompaana) [inline setup.md:130]:
```suggestion
3. Update the Terraform section of [1-preprovision/main.tf](./1-preprovision/main.tf) with your values for `resource_group_name`, `container_name`, and `storage_account_name`. Leave the key as `**"magicaks-longlasting"**`.
```

File: setup.md [hunk 32]
```diff
@@ -0,0 +1,304 @@
...
+        backend "azurerm" {
+            resource_group_name  = "longlasting"
+            container_name = "tfstate"
+            key = "magicaks-longlasting"
+            storage_account_name = "longlasting"
+        }
+    }
+    ```
+
+### 7. Set up the environment variables for terraform
```
Comment (tompaana) [inline setup.md:143]:
```suggestion
### 7. Set up the environment variables for Terraform
```

**Review by tompaana — CHANGES_REQUESTED** [review]

File: setup.md [hunk 33]
```diff
@@ -0,0 +1,304 @@
...
+> **IMPORTANT:** There is a cost associated with provisioning resources on Azure. Specifically the Azure Firewall costs by the hour. If you are setting this up for testing purposes, you may want to tear down all resources after testing to avoid high bills for resources you don't use.
+
+## Software requirements
+
+We have tested this repo with OSX, Linux and Windows (on [WSL](https://docs.microsoft.com/en-us/windows/wsl/)).
+
+You need to install:
+
+* [curl](https://curl.se/)
+* [kubectl min version 1.18.1](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
```
[the remark this answers is not in the document]
Comment (TessFerrandez) [reply setup.md]: Removed kubectl - per Anders comment that we are not using it anymore - so skipping this

File: setup.md [hunk 34]
```diff
@@ -0,0 +1,304 @@
+# Magic AKS Setup
+
+> **IMPORTANT:** There is a cost associated with provisioning resources on Azure. Specifically the Azure Firewall costs by the hour. If you are setting this up for testing purposes, you may want to tear down all resources after testing to avoid high bills for resources you don't use.
+
+## Software requirements
+
+We have tested this repo with OSX, Linux and Windows (on [WSL](https://docs.microsoft.com/en-us/windows/wsl/)).
```
[the remark this answers is not in the document]
Comment (TessFerrandez) [reply setup.md:7]: will fix

File: setup.md (hunk 3 above)
Comment (Lybecker) [reply setup.md:108]: I don't think you want a random storage account name. Then the TerraForm state will be in different location everytime you run the script?

File: setup.md [hunk 35]
```diff
@@ -0,0 +1,295 @@
...
+   4. RBAC enabled.
+   5. Calico as the network plugin
+   6. Kubernetes version = 1.19.7
+
+2. Flux gitOps operator
+3. Azure KeyVault
+4. akv2k8s operator installed to provide seamless access to KeyVault secrets.
+5. Service bus integrated and primary connection string stored in KeyVault and exposed in cluster as K8s secret.
+6. Integration with Azure Active Directory for K8s RBAC.
+7. Pod security policies enabled and a restricted policy added.
```
Comment (tompaana) [inline setup.md:293]:
```suggestion
* Pod security policies enabled and a restricted policy added
```

Comment (tompaana) [reply setup.md:293]: Hold on a minute, buster! Just before you told me that "AKS has deprecated PSP in favor of Azure Policy". What kind of shenanigans are you pulling here?

File: setup.md [hunk 36]
```diff
@@ -0,0 +1,295 @@
...
+8. Azure Policy enabled on the cluster. No policies assigned right now.
```
Comment (tompaana) [inline setup.md:294]:
```suggestion
* Azure Policy enabled on the cluster (no policies assigned right now)
```

File: setup.md [hunk 37]
```diff
@@ -0,0 +1,295 @@
...
+9. Azure Firewall integrated with network and application rules as recommended by AKS.
```
Comment (tompaana) [inline setup.md:295]:
```suggestion
* Azure Firewall integrated with network and application rules as recommended by AKS
```

File: setup.md [hunk 38]
```diff
@@ -0,0 +1,295 @@
...
+10. Grafana connected to log analytics workspace of the cluster is running in Azure container instance backed by managed Postgresql Azure database.
```
Comment (tompaana) [inline setup.md:296]:
```suggestion
* Grafana connected to Log Analytics workspace of the cluster is running in Azure Container Instances backed by managed PostgreSQL database
```

File: setup.md [hunk 39]
```diff
@@ -0,0 +1,295 @@
...
+
+Congratulations, you have provisioned your AKS cluster with the following resources
+
+1. AKS cluster.
+
+   1. VMSS node pool with 1 - 5 nodes.
+   2. Container insights in enabled.
+   3. ~~Pod security policies enabled.~~ (AKS has deprecated PSP in favor of Azure Policy)
+   4. RBAC enabled.
+   5. Calico as the network plugin
```
Comment (tompaana) [inline setup.md:285]:
```suggestion
   * Calico as the network plugin
```

File: setup.md [hunk 40]
```diff
@@ -0,0 +1,295 @@
...
+   6. Kubernetes version = 1.19.7
```
Comment (tompaana) [inline setup.md:286]:
```suggestion
   * Kubernetes version = 1.19.7
```

**Review by tompaana — CHANGES_REQUESTED** [review]

File: setup.md [hunk 41]
```diff
@@ -0,0 +1,295 @@
...
+    ```
+
+## What have we installed?
+
+Congratulations, you have provisioned your AKS cluster with the following resources
+
+1. AKS cluster.
+
+   1. VMSS node pool with 1 - 5 nodes.
+   2. Container insights in enabled.
```
[the remark this answers is not in the document]
Comment (TessFerrandez) [reply setup.md]: will add this

File: setup.md (hunk 35 above)
Comment (TessFerrandez) [reply setup.md]: I'll add an issue about this

**Review by tompaana — APPROVED** [review]

**Pull request merged**